[Chore] Separate out optional dependency checks from vllm.utils

dongbo910220 opened the pull request:
Part of #26900

This PR continues the effort to clean up vllm.utils by consolidating the helper functions used for checking optional dependencies into vllm/utils/import_utils.py.

Following @yewentao256's suggestion, these functions have been merged into import_utils.py since they all depend on importlib.util.find_spec, making it more logical to group all import-related utilities together.

 The following functions have been moved:

  vllm.utils.has_pplx -> vllm.utils.import_utils.has_pplx
  vllm.utils.has_deep_ep -> vllm.utils.import_utils.has_deep_ep
  vllm.utils.has_deep_gemm -> vllm.utils.import_utils.has_deep_gemm
  vllm.utils.has_triton_kernels -> vllm.utils.import_utils.has_triton_kernels
  vllm.utils.has_tilelang -> vllm.utils.import_utils.has_tilelang

CC: @DarkLight1337

### Test Plan

### Test Result

dongbo910220 committed:
refactor(utils): extract optional dependency checks to dedicated module

Move optional dependency detection functions to vllm/utils/optional_deps.py:
- has_pplx: Check for pplx_kernels package
- has_deep_ep: Check for deep_ep package
- has_deep_gemm: Check for deep_gemm package
- has_triton_kernels: Check for triton_kernels package
- has_tilelang: Check for tilelang package
- _has_module: Cached helper for module detection

This reduces vllm/utils/__init__.py from 1310 to 1285 lines and
groups related dependency checking utilities in a single module.

Contributes to #26900

Signed-off-by: dongbo910220 <1275604947@qq.com>

yewentao256 commented on benchmarks/overheads/benchmark_hashing.py:
```diff
@@ -7,7 +7,6 @@
 from vllm import LLM, SamplingParams
 from vllm.utils import FlexibleArgumentParser
 
-# A very long prompt, total number of tokens is about 15k.
```
same here

yewentao256 reviewed (commented):
Thanks for the work!
I think this should be put inside the `import_utils` since they all depends on `importlib.util.find_spec(module_name)`

dongbo910220 committed:
refactor(utils): merge optional dependency checks into import_utils

Move optional dependency detection functions (has_pplx, has_deep_ep,
has_deep_gemm, has_triton_kernels, has_tilelang) from optional_deps.py
into import_utils.py. This consolidates all import-related utilities
into a single module since they all depend on importlib.util.find_spec.

Changes:
- Add optional dependency detection functions to import_utils.py
- Update all imports from vllm.utils.optional_deps to vllm.utils.import_utils
- Remove vllm/utils/optional_deps.py
- Restore accidentally deleted comments in benchmark and example files

Contributes to #26900

Signed-off-by: dongbo910220 <1275604947@qq.com>

yewentao256 reviewed (approved):
LGTM, thanks for the work!

dongbo910220 committed:
Update vllm/utils/import_utils.py

Co-authored-by: Wentao Ye <44945378+yewentao256@users.noreply.github.com>
Signed-off-by: dongbo910220 <32610838+dongbo910220@users.noreply.github.com>

dongbo910220 committed:
style: fix spacing in import_utils.py to comply with PEP 8

Add required blank lines between class definition and module-level
comment as per PEP 8 style guide (two blank lines between top-level
definitions).

Signed-off-by: dongbo910220 <1275604947@qq.com>

dongbo910220 committed:
chore(mxfp4): update has_triton_kernels import to use import_utils

Update import path from `vllm.utils` to `vllm.utils.import_utils` to
align with the refactoring that merged optional dependency checks into
import_utils module.

This completes the migration for has_triton_kernels references across
the codebase.

Signed-off-by: dongbo910220 <1275604947@qq.com>

dongbo910220 committed:
Merge branch 'main' into feature/refactor-optional-deps

Resolve conflict in vllm/model_executor/layers/fused_moe/layer.py by
combining imports from both branches:
- Keep import_utils imports (has_deep_ep, has_pplx) from our branch
- Add current_stream import from main branch

Signed-off-by: dongbo910220 <1275604947@qq.com>

dongbo910220 committed:
fix(utils): remove unused importlib import

The importlib module was imported but never used in __init__.py after
the optional dependency checks were moved to import_utils.py.

Fixes ruff F401 error.

Signed-off-by: dongbo910220 <1275604947@qq.com>

DarkLight1337 committed:
Merge branch 'main' into feature/refactor-optional-deps

DarkLight1337 committed:
Merge branch 'main' into feature/refactor-optional-deps

yewentao256 reviewed (approved):
Since All CI pass, let's get this landed
